### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by Aloso**
This PR removes some string allocations in the `supports_color` function.

I originally wanted to use `Result::as_deref` for this, but since I wasn't sure what the minimum supported Rust version is, and `Result::as_deref` is only available since 1.47, I just wrote a helper function instead.

Other changes:

- I removed `min` since it is guaranteed to be 0 everywhere where it is used
- I reordered some code paths to do less work by returning earlier if `FORCE_COLOR` or `NO_COLOR` is set
- I fixed the tests, which were flaky; cargo runs tests in parallel, which does not work properly if multiple tests read and write environment variables. The solution was to add a lock that ensures the tests run sequentially.

### CODE REVIEW

**Review by zkat — APPROVED** [review]
This is awesome, and a welcome change! Thanks a lot!

File: src/lib.rs [hunk 1]
```diff
@@ -61,35 +77,27 @@ fn translate_level(level: usize) -> Option<ColorLevel> {
 
 fn supports_color(stream: Stream) -> usize {
     let force_color = env_force_color();
-    let no_color = match std::env::var("NO_COLOR") {
-        Ok(val) if val == *"0" => false,
-        Ok(_) => true,
-        Err(_) => false,
-    };
-    let min = std::cmp::max(force_color, 0);
```
Comment (zkat) [inline src/lib.rs:69]: wait, I have a question. I think I'm missing something. Why is this guaranteed to be 0 anywhere where it's used?

Comment (zkat) [reply src/lib.rs:69]: ah nevermind. Literally the first `if` clause lol.

**Comment by zkat** [comment]
published as `supports-color@1.3.1` :)

**Comment by Aloso** [comment]
Thank you!

**Pull request merged**